src/System.IO.Compression/src/System/IO/Compression/DeflateManaged/InflaterManaged.cs

```diff
@@ -211,6 +214,7 @@ private bool Decode()
                 }
                 else if (_state == InflaterState.StartReadingFooter || _state == InflaterState.ReadingFooter)
                 {
+                    Debug.Assert(_formatReader != null);
```

Comment:
Instead of adding two `Debug.Assert` you can add one inside the block of `if (_hastFormatReader)` before the `if (_state == InflaterState.ReadingHeader)`